lib/ansible/modules/network/cumulus/nclu.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016-2017, Cumulus Networks <ce-ceng@cumulusnetworks.com>
+#
+# This file is part of Ansible
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+DOCUMENTATION = '''
+---
+module: nclu
+version_added: "2.2.3"
+author: "Cumulus Networks"
+short_description: Configure network interfaces using NCLU
+description:
+    - Interface to the Network Command Line Utility, developed to make it easier
+      to configure operating systems running ifupdown2 and Quagga, such as
+      Cumulus Linux. Command documentation is available at
+      https://docs.cumulusnetworks.com/display/DOCS/Network+Command+Line+Utility
+options:
+    commands:
+        description:
+            - A list of strings containing the net commands to run.
+    template:
+        description:
+            - A single, multi-line string with jinja2 formatting. This string
+              will be broken by lines, and each line will be run through net.
+              Mutually exclusive with 'commands'.
+    commit:
+        description:
+            - When true, performs a 'net commit' at the end of the block.
+    abort:
+        description:
+            - Boolean. When true, perform a 'net abort' before the block.
+              This cleans out any uncommitted changes in the buffer.
+    atomic:
+        description:
+            - When true, equivalent to both commit and abort being true.
+    description:
+        description:
+            - Commit description that will be recorded to the commit log if
+              commit or atomic are true.
+'''
+
+EXAMPLES = '''
+
+## Add two interfaces without committing any changes
+
```

Comment:
Examples should include a ` - name:`, see http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#examples-block for details.